src/adapter/src/session/vars.rs

```diff
@@ -1193,7 +1193,7 @@ impl SystemVars {
     /// Returns an iterator over the configuration parameters and their current
     /// values on disk.
     pub fn iter(&self) -> impl Iterator<Item = &dyn Var> {
-        let vars: [&dyn Var; 22] = [
+        let vars: [&dyn Var; 23] = [
```

Comment:
If there is a way to not declare the type or use some type inference magic please feel free to force-push!

Reply:
@parker-timmerman any thoughts?

Reply:
Ideally we could do `let vars: [&dyn Var; _] = ...` and it would infer the length of the array, unfortunately that syntax is gated behind an unstable feature [generic_arg_infer](https://github.com/rust-lang/rust/issues/85077).

What we could do though is put `&dyn Var` on the first element i.e.
```
let vars = [
    &self.config_has_synced_once as &dyn Var,
    &self.max_aws_privatelink_connections,
    ...
];
```
which tells the compiler that we want to treat these as `&dyn Var`, and allows it to infer the length.

It doesn't look great, but it's the only way I know of to get the type inference here

Reply:
tbh I'm surprised the compiler didn't complain about the array being the wrong length 🤔

Reply:
I really like that suggestion. This is not the first time this exact merge skew issue has happened, and it probably won't be the last.

Reply:
I tried the `_` placeholder but the compiler didn't like it.